components/roadmap/RoadmapItem.tsx

```diff
@@ -0,0 +1,154 @@
+import { useState } from 'react';
+
+import IconArrowRight from '../icons/ArrowRight';
+import Modal from '../Modal';
+/* eslint-disable import/no-cycle*/
+import RoadmapList from './RoadmapList';
+
+interface IPillProps {
+  item: {
+    done?: boolean;
+    url?: string;
+    description?: string;
+    title: string;
+  };
+  colorClass?: string;
+  isCollapsible?: boolean;
+  isCollapsed?: boolean;
+  onClickCollapse?: () => void;
+}
+
+/**
+ * @description Pill is a component to display a roadmap item.
+ * @param {object} props.item - The roadmap item.
+ * @param {boolean} props.item.done - Whether the item is done.
+ * @param {string} props.item.url - The URL associated with the item.
+ * @param {string} props.item.description - The description of the item.
+ * @param {string} props.item.title - The title of the item.
+ * @param {string} props.colorClass - The color class for styling.
+ * @param {boolean} props.isCollapsible - Whether the item is collapsible.
+ * @param {boolean} props.isCollapsed - Whether the item is collapsed.
+ * @param {function} props.onClickCollapse - Function to handle click on collapse.
+ */
+function Pill({
```

Comment:
Create a separate file for this component as it should be handled separately, looking into the depth of code it has.